src/libhost1x/host1x-gr3d.c

```diff
@@ -85,6 +142,8 @@ static int host1x_gr3d_reset(struct host1x_gr3d *gr3d)
 	uint32_t fence;
 	int err;
 
+	host1x_gr3d_reset_hw();
```

Comment:
Absolutely! However, it's very convenient to have it now in grate, since it's relatively easy to hang the GPU and rebooting each time it happens is annoying.